### ISSUE TITLE
[CPU][CUDA] `cpu` and `cuda` implementations of `log_softmax` backward seem to disagree on `dim > 2`

### ISSUE THREAD

**Issue by eqy** [pytorch/pytorch#155043]
### 🐛 Describe the bug

input dim > 2 seems to result in CPU setting grad to all zeros
```
import torch
import torch.nn.functional as F

dtype = torch.float
device = 'cuda'
logits = torch.randn(5, 1, 24, dtype=dtype, device=device, requires_grad=True)
out = F.log_softmax(logits, dim=-1)
grad = torch.randn_like(out)
out.backward(grad)
logits_cpu = logits.detach().cpu()
logits_cpu.requires_grad = True
out_cpu = F.log_softmax(logits_cpu, dim=1)
out_cpu.backward(grad.detach().cpu())
torch.testing.assert_close(out, out.cuda())
print(logits.grad, logits_cpu.grad)
torch.testing.assert_close(logits.grad, logits_cpu.grad.cuda())
```
discovered while checking cases for #154778

### Versions

2.8-dev

**Comment by narekmalk** [comment]
Can I work on this issue?

**Comment by narekmalk** [comment]
@eqy In your code, you're using `dim=1` in `F.log_softmax(logits_cpu, dim=1)`. Shouldn't this be `dim=-1` to match the CUDA case? When I use `dim=-1`, the results on CPU and CUDA match

**Comment by ZailiWang** [comment]
As the input shape is [5,1,24], the output of `out_cpu = F.log_softmax(logits_cpu, dim=1)` is expected to be all-zero. I also tested `dim=0 or 2` cases and the CPU/CUDA results are identical.

**Comment by eqy** [comment]
Ah you're right, that's embarrassing.

Thanks and sorry for the noise!
